src/components/Stations.tsx

```diff
@@ -1,22 +1,73 @@
+        const url = baseUrl + 'originStation=' + departureStationChoice + '&destinationStation=' + arrivalStationChoice + trailingUrl;
 
-    useEffect(() => {
-        fetchStations()
-            .then((value) => console.log(value))
-            .catch((err) => console.log(err))
-            .finally(() => console.log('finally'));
-    }, []);
+        fetch(url, {
+            headers: {
+                'X-API-KEY': `${process.env.REACT_APP_X_API_KEY}`,
+            }, 
+        })
+            .then(response => response.json())
+            .then(data => data.numberOfChildren)
+            .then(value => alert(value))
+            .then(() => alert(url))
+            .catch(error => {
+                alert('Error fetching data:' + error);
+            })
+            .finally(() => console.log(url));
+    };
 
-    console.log('hello world');
+    const stationList = [
+        { id: 'PAD', name: 'Paddington' },
+        { id: 'LST', name: 'Liverpool Street' },
+        { id: 'KGX', name: 'Kings Cross' },
+        { id: 'KTN', name: 'Kentish Town' },
+        { id: 'LBG', name: 'London Bridge' },
+    ];
+
+    // let stationChoice: string;
+    const [departureStationChoice, setDepartureStationChoices] = useState('PAD');
+    const [arrivalStationChoice, setArrivalStationChoices] = useState('PAD');
+
+    // useEffect(() => {
+    //  fetchStations()
+    // .then(response => response.json())
+    // .then(list => list.stations[0].id)
+    // .then((value) => alert(value))
+    // .catch((err) => alert(err))
+    // .finally(() => alert('finally'));
+    // }, []);
 
     return (
-        <div>
-            Stations!
+        <div className = "station-wrapper">
+            <div className = "dropdown-selections">
+                <div>
+                    <label htmlFor = "departure stations">Select the departure station: </label>
+
+                    <select value = { departureStationChoice } name = "stations" id = "departure-stations" onChange = { (e) => setDepartureStationChoices(e.target.value) }> 
+                        { stationList.map( (stationObj: {id: string; name: string}, index: number) => (<Station key = { index } id = { stationObj.id } name = { stationObj.name }/>)) }
+                    </select>
+                </div>
+                <div>
+                    <label htmlFor = "arrival stations"> Select the arrival station: </label>
+
+                    <select value = { arrivalStationChoice } name = "stations" id = "arrival-stations" onChange = { (e) => setArrivalStationChoices(e.target.value) }> 
+                        { stationList.map( (stationObj: {id: string; name: string}) => (<Station key = { stationObj.id } id = { stationObj.id } name = { stationObj.name }/>)) }
+                    </select>
+                </div>
+            </div>
+            <button className = "button" type = "button" id = "submitStation" onClick = { handleOnClick } >Submit</button>
```

Comment:
we don't need an id field on the button here.

Similarly on the select we don't need name or id.

We sometimes need to add an id to make testing easier but otherwise we shouldn't need it